lib/factory.rb

```diff
@@ -0,0 +1,77 @@
+class Factory
+  class << self
+    def new(*args, &block)
+      const_set(args.shift.capitalize, class_new(*args, &block)) if args.first.is_a?(String)
+      class_new(*args, &block)
+    end
+
+    def class_new(*args, &block)
+      Class.new do
+        attr_accessor(*args)
+
+        def initialize(*vars)
+          raise ArgumentError, "Expected #{args.count} argument(s)" if args.count != vars.count
+
+          args.each_index do |index|
+            instance_variable_set("@#{args[index]}", vars[index])
+          end
+        end
+
+        define_method :args do
+          args
+        end
+
+        def [](arg)
+          arg.is_a?(Integer) ? values[arg] : instance_variable_get("@#{arg}")
+        end
+
+        def []=(arg, value)
+          instance_variable_set("@#{arg}", value)
+        end
+
+        def each(&block)
+          values.each(&block)
+        end
+
+        def each_pair(&pair)
+          to_h.each_pair(&pair)
+        end
+
+        def dig(*args)
+          args.reduce(to_h) { |memo, arg| (memo[arg].is_a? NilClass) ? (return nil) : memo[arg] }
```

Comment:
don't copy not fully correct methods from other students repos.
https://github.com/WoLkErSs/Factory/pull/1/files#diff-404248b357b047b27fceb12341cece15R54